Framework/Crystal/src/IndexPeaks.cpp

```diff
@@ -412,6 +412,23 @@ std::map<std::string, std::string> IndexPeaks::validateInputs() {
     helpMsgs[Prop::PEAKSWORKSPACE] = exc.what();
   }
 
+  // get all runs which have peaksin the table
+  bool commonUB = this->getProperty(Prop::COMMONUB);
+  if (commonUB) {
+    auto &allPeaks = ws->getPeaks();
+    std::unordered_map<int, int> peaksPerRun;
+    auto it = allPeaks.begin();
```

Comment:
Can we use `allPeaks.cbegin()` to get a const iterator? You'll need to use `.cend` in the `while` test too.